recipes/default.rb

```diff
@@ -67,63 +136,77 @@
 	end
 end
 
-if node[:platform_version].to_i == 6
-	if node['authconfig']['ldap']['enable']
-		package 'pam_ldap' do
-			action :install
-		end
-	end
+# SSSD configuration if installed
+if sssd_action == 'install'
 
-	package "sssd" do
-		action :install
-	end
-
-	service "sssd" do
-		supports :status => true, :restart => true, :reload => true
-		# Avoid starting or restarting sssd if disabled,
-		# especially when kerberos is enabled, and ldap not
-		restart_command "/sbin/chkconfig sssd --list | grep -v :on || /sbin/service sssd restart"
-		start_command "/sbin/chkconfig sssd --list | grep -v :on || /sbin/service sssd start"
-	end
+  execute 'restorecon /etc/sssd/sssd.conf' do
+    action :nothing
+  end
 
 	execute "clean_sss_db" do
 		command "rm -f /var/lib/sss/db/*"
 		action :nothing
 	end
 
-	execute "restorecon /etc/sssd/sssd.conf" do
-		action :nothing
-	end
+  directory '/etc/sssd' do
+    mode 0600
+    owner 'root'
+    group 'root'
+    action :create
+  end
 
-	template "/etc/sssd/sssd.conf" do
-		source "sssd.conf.erb"
+ 	template '/etc/sssd/sssd.conf' do
+ 		source 'sssd.conf.erb'
 		mode 0600
-		owner "root"
-		group "root"
-		notifies :run, "execute[clean_sss_db]", :immediately
-		notifies :run, "execute[restorecon /etc/sssd/sssd.conf]", :immediately
-		notifies :restart, "service[sssd]", :immediately
-		notifies :reload, "ohai[reload]", :immediately
-	end
+ 		owner 'root'
+ 		group 'root'
+ 		notifies :run, 'execute[clean_sss_db]', :immediately
+ 		notifies :run, 'execute[restorecon /etc/sssd/sssd.conf]', :immediately if node['platform_version'].to_f >= 6
+ 		notifies :restart, 'service[sssd]', :delayed
+ 		notifies :reload, 'ohai[reload_passwd]', :immediately
+ 	end
+end
 
-elsif node[:platform_version].to_i == 5
+service "sssd" do
+  supports :status => true, :restart => true, :reload => true
+  # Avoid starting or restarting sssd if disabled,
+  # especially when kerberos is enabled, and ldap not
+  restart_command "/sbin/chkconfig sssd --list | grep -v :on || /sbin/service sssd restart"
```

Comment:
This no longer works as expected with Fedora 25 and other systemd based recent systems.  "/sbin/chkconfig sssd --list | grep -v :on" always succeeds now, because the daemon is instead managed by systemd and not reported by /sbin/chkconfig. That would take a systemctl command to detect as being enabled.

I don't think it's a loss: the "authconfig" command itself automatically restarts sssd if appropriate, so the restart command here is extraneous in most circumstances. But just a heads up, if commands are edited by the authconfig cookbook that demand an sssd restart, it won't happen on recent systemd based Linux releases.